plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScannerTest.java

```diff
@@ -43,25 +47,34 @@
 public class WebXmlScannerTest {
 
   private static IFile resource;
-  private static IProject project;
-  private WebXmlScanner scanner = new WebXmlScanner();
+  private static IFile servletClass;
+  private static IJavaProject javaProject;
+  private WebXmlScanner scanner;
   
-  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator();
+  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator().withClasspathContainerPath("src/main/java");
   
   @BeforeClass
   public static void setUpBeforeClass() throws CoreException {
-    project = projectCreator.getProject();
+    IProject project = projectCreator.getProject();
+    javaProject = JavaCore.create(project);
+    
+    IPackageFragmentRoot root = javaProject.getPackageFragmentRoot("src/main/java");
+    root.createPackageFragment("com.test", false, null);
```

Comment:
It turns out that you don't even need to do anything. The default Java source folder was the project root, and as such, you were already creating `ServletClass` within the package `src.main.java`. That is, `assertTrue(WebXmlScanner.classExists(javaProject, "src.main.java.ServletClass"));` succeeds.

Now, there does exist a way to change the default Java source folder to something else, but this will required quite some effort. Not sure if it's worth setting up the project that way.